CorsixTH/Lua/dialogs/edit_room.lua

```diff
@@ -1086,72 +1114,96 @@ local function checkDoorWalls(x, y, wall, has_swingdoor)
 --!param xpos (int) X position of the tile.
 --!param ypos (int) Y position of the tile.
 --!param player_id (int) Player id owning the hospital.
---!param flag_names (array) If set, array with two additional required properties.
+--!param wall wall facing direction
+--!param wall (string) wall facing direction
+--!param world wall facing direction
 --!return Whether the tile is considered to be valid.
-local function validDoorTile(xpos, ypos, player_id, flag_names)
+local function validDoorTile(xpos, ypos, player_id, wall, world)
   local th = TheApp.map.th
-
   local tile_flags = th:getCellFlags(xpos, ypos)
-  if not (tile_flags.buildable or tile_flags.passable or tile_flags.owner == player_id) then return false end
-  if not flag_names then return true end
-  return tile_flags[flag_names[1]] and tile_flags[flag_names[2]]
+  if not (tile_flags.buildable and tile_flags.owner == player_id) then
+    -- any object will cause it to be blocked (ignore litter)
+    if not(tile_flags.thob == 0 or tile_flags.thob == 62) then return false end
+    if tile_flags.passable then
+      -- check for any object with a passable tile built within the door footprint
+      -- drinks machine and reception desk in corridors for example
+      for o in pairs(world:findAllObjectsNear(xpos, ypos, 1)) do
+        if o.object_type.id ~= "bench" then
+          for _, xy in ipairs(o:getWalkableTiles()) do
+            if xy[1] == xpos and xy[2] == ypos then
+              return false
+            end
+          end
+        end
+      end
+    end
+  end
+  return (tile_flags.thob == 0 or tile_flags.thob == 62)
 end
 
 function UIEditRoom:setDoorBlueprint(orig_x, orig_y, orig_wall)
-  local x = orig_x
-  local y = orig_y
-  local wall = orig_wall
-
-  -- Used to get the adjacent tiles when placing swing doors.
-  local x_mod
-  local y_mod
-  if wall == "south" then
-    y = y + 1
-    wall = "north"
-    x_mod = 2
-  elseif wall == "east" then
-    x = x + 1
-    wall = "west"
-    y_mod = 2
-  elseif wall == "north" then
-    x_mod = 2
-  else
-    y_mod = 2
+  local function doorWallOffsetCalculations(x, y, wall)
+    local x_mod
+    local y_mod
+    if wall == "south" then
+      y = y + 1
+      wall = "north"
+      x_mod = 2
+    elseif wall == "east" then
+      x = x + 1
+      wall = "west"
+      y_mod = 2
+    elseif wall == "north" then
+      x_mod = 2
+    else
+      y_mod = 2
+    end
+    return x, y, x_mod, y_mod, wall
   end
+  
+  local x, y, x_mod, y_mod, wall = doorWallOffsetCalculations(orig_x, orig_y, orig_wall)
   local map = TheApp.map.th
 
   if self.blueprint_door.anim then
     if self.room_type.swing_doors then
       if self.blueprint_door.anim[1] then
+        -- retrieve the old door position details to reset the blue print
+        local oldx, oldy, oldx_mod, oldy_mod, old_wall = doorWallOffsetCalculations(self.blueprint_door.floor_x,
+                                                           self.blueprint_door.floor_y,
```

Comment:
Line continuations are done by a double indent on the line below. That avoids that code shifts all the way to the right margin, creating many lines with mostly white space.